.github/workflows/publish_imgs.yml

```diff
@@ -0,0 +1,155 @@
+# nix-garage GitHub Actions Workflow for publishing images to Docker Hub
+
+### Workflow
+# 1. We want this to be ran manually i.e. with a comment in a PR. To do this, we can have a release process where we update what is currently out in prod via a PR
+# 2. Once the PR is made, we then have someone (who is an approver) comment in the PR "Publish Images" (WIP)
+# 3. GitHub Actions will detect this comment being made, and check if the comment was made by an approved person, alongside what they commented.
+# 4. If it matches, we fire off this job, where we log into docker.io with nwiauto (also pass in the password via a secret declared in the repo)
+# 5. Once we logged in, we then execute the script "publish-imgs", which will publish the image passed into it.
+
+name: Publish Docker Images
+
+on:
+  # This job will be fired off when a comment has been made in an issue (which also impliitly includes PRs)
+  issue_comment:
+    types: [created]
+
+jobs:
+  build:
+    name: Publish Image to Dockerhub
+    
+    # Running on a GitHub hosted runner
+    runs-on: ubuntu-18.04
+
+    # Checks if the comment contains the word 'publish'
+    # As well as if the comment was made in a PR
+    if: > 
+      startsWith(github.event.comment.body, 'publish')
+      && startsWith(github.event.issue.pull_request.url, 'https://')
+
+    steps:
+      # Checks out this repository to the runner
+      - uses: actions/checkout@v2
+
+      # Installs pre-reqs the runner does not initially have
+      # TODO: This can be cached in the future
+      - name: Install Pre-Reqs
+        id: prereq
+        run: |
+          set -x
+          curl -o install-nix-2.3.6 https://releases.nixos.org/nix/nix-2.3.6/install
+          curl -o install-nix-2.3.6.asc https://releases.nixos.org/nix/nix-2.3.6/install.asc
+          gpg2 --recv-keys B541D55301270E0BCF15CA5D8170B4726D7198DE
+          gpg2 --verify ./install-nix-2.3.6.asc
+          sh ./install-nix-2.3.6
+
+          . /etc/os-release
+          sudo sh -c "echo 'deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/x${NAME}_${VERSION_ID}/ /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list"
+          wget -nv https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/x${NAME}_${VERSION_ID}/Release.key -O- | sudo apt-key add -
+          sudo apt-get update -qq
+          sudo apt-get install skopeo
+        continue-on-error: false
+
+      # Verifies if the commenter is an approver, otherwise, the job will output an error
+      - name: Verify that Commenter is Approver
+        id: verify
+        run: |
+          resp=$(curl -sSf \
+          --url https://api.github.com/orgs/Nebulaworks/teams/approvers/memberships/${{ github.event.comment.actor.login }} \
+          --header 'Authorization: Bearer ${{ secrets.NWIAUTO_PAT }}' \
```

Comment:
Ah you raise some good points:

I really dont mind either way for posts from `nwiauto `or the `github-actions-bot`. But I will say that if we could leverage the `GITHUB_TOKEN` for all of this actions functionality it will get us out of managing an additional secret which is definitely a ➕.

So why dont you see if you can get endpoint working for the teams and if so well go with the `github-actions-bot` via the `GITHUB_TOKEN`. If it does not work for the teams endpoint then well keep it as you initial had it with the `NWIAUTO_PAT`. Sound good?

Reply:
Cool! I did some more testing with other API endpoints, and unfortunately, the `GITHUB_TOKEN` does not work in getting those endpoints I mentioned earlier. At this point, I think sticking with the PAT would be best, even though it is another secret to manage.

Reply:
Sounds good, thanks for giving it a test 💯